lib/matplotlib/axes/_axes.py

```diff
@@ -3822,6 +3828,13 @@ def apply_mask(arrays, mask):
 
         self._request_autoscale_view()
         caplines = caplines['x'] + caplines['y']
+
+        if(elinestyle is not None):
+            if(xerr is not None and yerr is not None): 
+                barcols[-2].set_linestyle(elinestyle)
+                barcols[-1].set_linestyle(elinestyle)
```

Comment:
Instead of modifying this here afterwards, please set the style in `eb_lines_style` (around line 3700).